lib/OpenQA/CacheService/Client.pm

```diff
@@ -71,6 +73,10 @@ sub enqueue ($self, $request) {
     return undef;
 }
 
+sub _is_ipv4_addr ($ip) {
+    return $ip =~ /^(?<!\d)(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})(?!\d)$/;
```

Comment:
Why the `(?<!\d)` lookbehind? The anchor `^` at the beginning would make that impossible anyway, right?
Or can you give an example?
Same for the `(?!\d)` lookahead.

Reply:
An alternative:
```perl
use Regexp::Common qw(net);
return $ip =~ m/$RE{net}{IPv4}/;
```

Reply:
fixed.